tasks/main.yml

```diff
@@ -1,38 +1,71 @@
 ---
-  - name: Installing SSSD and Kerberos dependencies using yum
+  - name: Installing Realmd, SSSD, and Kerberos dependencies using yum
     yum:
-        name: {{ linux_adbinding__dependencies_yum }}
-        state: present
-    when: ansible_pkg_mgr == yum
+      name: "{{ linux_adbinding__dependencies_yum }}"
+      state: present
+    when: ansible_pkg_mgr == "yum"
 
-  - name: Installing SSSD and Kerberos dependencies using apt
-    yum:
-        name: {{ linux_adbinding__dependencies_apt }}
-        state: present
-    when: ansible_pkg_mgr == apt
+  - name: Installing Realmd, SSSD, and Kerberos dependencies using apt
+    apt:
+      name: "{{ linux_adbinding__dependencies_apt }}"
+      state: present
+    when: ansible_pkg_mgr == "apt"
+
+  - name: Run Realm discover if not specified yet
+    shell: /bin/bash -c "/usr/sbin/realm discover | grep "domain-name" | head -n 1 | sed 's/.*:\s*//'"
+    register: realm_discover_output
+    changed_when: false
+    when: linux_adbinding__domain is undefined or linux_adbinding__domain == ''
+
+  - name: set fact for discovered variable
+    set_fact:
+      linux_adbinding__domain: "{{ realm_discover_output.stdout }}"
+    when: linux_adbinding__domain is undefined or linux_adbinding__domain == ''
+
+  - debug:
+      msg: "Specified or discovered domain is: {{ linux_adbinding__domain }}"
 
-  - name: Creating SSSD configuration
+  - name: fail if nothing specified or discovered
+    assert:
+      that:
+        - linux_adbinding__domain != ""
+
+  - name: Creating Realmd configuration
     template:
-        src: "sssd.conf.j2"
-        dest: "/etc/sssd/sssd.conf"
-        mode: 0600
-        owner: root
-        group: root
-    notify:
-    - restart sssd
+      src: "realmd.conf.j2"
+      dest: "/etc/realmd.conf"
+      mode: 0644
+      owner: root
+      group: root
 
   - name: Creating Kerberos configuration
     template:
-        src: "krb5.conf.j2"
-        dest: "/etc/krb5.conf"
-        mode: 0644
-        owner: root
-        group: root
+      src: "krb5.conf.j2"
+      dest: "/etc/krb5.conf"
+      mode: 0644
+      owner: root
+      group: root
 
   - name: Joining AD
-    command: /bin/bash -c "echo -n {{ linux_adbinding__password}} | /usr/sbin/adcli join -U {{ linux_adbinding__user }} --stdin-password -O {{ linux_adbinding__ou }} {{ linux_adbinding__domain }}"
+    command: /bin/bash -c "echo -n {{ linux_adbinding__password}} | /usr/sbin/realm join -U {{ linux_adbinding__user }} {{ linux_adbinding__domain }}"
```

Comment:
This breaks Centos6 Compatibility which is fine for us, since we are fading it out.
We will change the supported platforms in the Ansible Galaxy Metadata accordingly.

Reply:
Yes, Centos 6 is almost completely phased out for us also. Thanks for understanding. I hadn't actually tried it, I was hoping Realm might install on 6 still.